packages/server/utils/tiptap/hocusPocusRedisPublisher.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  type afterLoadDocumentPayload,
+  type afterUnloadDocumentPayload,
+  type Extension
+} from '@hocuspocus/server'
+import {unpack} from 'msgpackr'
+import {SubscriptionChannel} from '../../../client/types/constEnums'
+import {CipherId} from '../CipherId'
+import publish from '../publish'
+import type {PublicPageNotificationPayload} from '../publishPageNotification'
+import RedisInstance from '../RedisInstance'
+
+export class RedisPublisher implements Extension {
+  priority = 1001
+
+  sub: RedisInstance
+
+  public constructor() {
+    this.sub = new RedisInstance('publicPage_sub')
+  }
+
+  private getChannelName(documentName: string) {
+    const [dbId] = CipherId.fromClient(documentName)
+    return `publicPage:${dbId}`
+  }
+  async afterLoadDocument({documentName, document}: afterLoadDocumentPayload) {
```

Comment:
when a document gets loaded on a hocus pocus server, it starts listening. when the doc gets unloaded (no more users are subscribed to changes) we stop listening.

i broke this out into an extension just for readability